src/components/DevToolsUi.tsx

```diff
@@ -5,80 +5,125 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
 import { DiagnosticData } from '@/types/DiagnosticData'
 
 import { ConsoleOutput } from './ConsoleOutput'
-import { CreateGithubIssue } from './CreateGithubIssue'
+import { DescriptionInfo } from './DescriptionInfo'
+import { Header } from './Header'
 import { LogsTable } from './LogsTable'
-import { MetadataTable } from './MetadataTable'
-import { QueryResults } from './QueryResults'
+import { QueryResultsPanel } from './QueryResultsPanel'
 import { RawContent } from './RawContent'
 
 interface DevToolsUIProps {
   diagnosticData: DiagnosticData
   slackFileId?: string
 }
 
+interface TabHeaderProps {
+  id: string
+  title: string
+  count: number | undefined
+}
+
+const TabHeader = ({ id, title, count }: TabHeaderProps) => {
+  return (
+    <TabsTrigger value={id} className="tabs-trigger">
+      {title}
+      <Badge variant="destructive" className="ml-2">
+        {count}
+      </Badge>
+    </TabsTrigger>
+  )
+}
+
 export const DevToolsUI = ({ diagnosticData, slackFileId }: DevToolsUIProps) => {
-  const [frontendErrorCount, setFrontendErrorCount] = useState(0)
+  const [frontendErrorCount, setFrontendErrorCount] = useState(
+    diagnosticData.frontendErrors?.length
+  )
 
-  const jsonString = JSON.stringify(diagnosticData, null, 2)
+  // Since quite a bit is optional, be smart about what to show first based on importance and fallback to the raw content
+  const initialTab =
+    diagnosticData.backendErrors && diagnosticData.backendErrors.length > 0
+      ? 'backendErrors'
+      : diagnosticData.bugReportDetails
+        ? 'details'
+        : diagnosticData.entityInfo
+          ? 'entity'
+          : 'raw'
 
   return (
     <div className="flex flex-col h-screen bg-background text-foreground">
-      <CreateGithubIssue diagnosticData={diagnosticData} slackFileId={slackFileId} />
+      <Header diagnosticData={diagnosticData} slackFileId={slackFileId} />
+      <DescriptionInfo diagnosticData={diagnosticData} />
       <div className="w-full p-4 overflow-auto relative">
-        <Tabs defaultValue="basicInfo" className="w-full">
+        <Tabs defaultValue={initialTab} className="w-full">
           <div className="w-full bg-background z-10">
             <TabsList className="tabs-list mb-4">
-              <TabsTrigger value="basicInfo">Basic Info</TabsTrigger>
-              <TabsTrigger value="entityInfo">Entity Info</TabsTrigger>
-              <TabsTrigger value="frontendErrors" className="tabs-trigger relative">
-                Console output
-                {frontendErrorCount > 0 && (
-                  <Badge variant="destructive" className="ml-2 absolute -top-2 -right-2 z-10">
-                    {frontendErrorCount}
-                  </Badge>
+              {diagnosticData.bugReportDetails && (
+                <TabsTrigger value="details">Details</TabsTrigger>
+              )}
+              {diagnosticData.backendErrors?.length > 0 && (
```

Comment:
these conditionals are begging to be DRYed up a bit. some of the common logic could be moved to the subcomponents, or at least extracted to booleans so we can be sure we get the same behavior for the tab and the tab content.

also, I'm not sure we necessarily want to hide these tabs at all. the fact that there are no errors in a tab is relevant and more likely to be surfaced if there's an empty tab than no tab at all

Reply:
I went with just checking for the data structure's existence instead so we hide if the structure is actively broken (which is unlikely but hey, we never know). That simplifies and I believe removes the need for extracting additional variables?